src/algebra/polynomial/group_ring_action.lean

```diff
@@ -0,0 +1,137 @@
+/-
+Copyright (c) 2020 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+import data.polynomial.monic
+import algebra.group_ring_action
+import algebra.group_action_hom
+
+/-!
+# Group action on rings applied to polynomials
+
+This file contains instances and definitions relating `mul_semiring_action` to `polynomial`.
+-/
+
+variables (M : Type*) [monoid M]
+
+namespace polynomial
+
+variables (R : Type*) [semiring R]
```

Comment:
This file just contains stuff moved from the other files where the import order no longer worked.

As a small change, `comm_semiring` is relaxed to `semiring` for some of these lemmas.